sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -92,17 +92,12 @@ EventImplPtr Scheduler::addCopyBack(Requirement *Req) {
   return NewCmd->getEvent();
 }
 
-Scheduler::~Scheduler() {
-  // TODO: Make running wait and release on destruction configurable?
-  // TODO: Process release commands only?
-  //std::lock_guard<std::mutex> lock(MGraphLock);
-  //for (GraphBuilder::MemObjRecord &Record : MGraphBuilder.MMemObjRecords)
-    //waitForRecordToFinish(&Record);
-  //MGraphBuilder.cleanupCommands([>CleanupReleaseCommands = <] true);
-}
+// The init_priority here causes the constructor for scheduler to run relatively
+// early, and therefore the destructor to run relatively late (after anything
+// else that has no priority set, or has a priority higher than 2000).
+Scheduler Scheduler::instance __attribute__((init_priority(2000)));
```

Comment:
Is initialization order problem solved somehow differently for Windows?

Reply:
I don't generally do Windows development, so I'm not very familiar with the linking rules on Windows.

My recollection of earlier tests I did was that just moving the global instance from function-local to global was sufficient to prevent the crashes I was seeing, since they were using a function-scope global instance (not created until first call) instead of a global-scope global instance (created on program startup).

Reply:
Please, make sure your patch doesn't break windows.

Reply:
I pushed a new version that makes it work on Windows, and should have the same effect as the GCC init_priority attribute.